src/lib/helpers/downloadFile.ts

```diff
@@ -7,27 +8,49 @@ export const download = (url: string, dest: string) =>
     fs.access(dest, fs.constants.F_OK, (err) => {
       if (err === null) reject('File already exists');
 
-      const request = https.get(url, (response) => {
-        if (response.statusCode === 200) {
-          const file = fs.createWriteStream(dest, { flags: 'wx' });
-          file.on('finish', () => resolve());
-          file.on('error', (err: any) => {
-            file.close();
-            if (err.code === 'EEXIST') reject('File already exists');
-            else fs.unlink(dest, () => reject(err.message)); // Delete temp file
-          });
-          response.pipe(file);
-        } else if (response.statusCode === 302 || response.statusCode === 301) {
-          //Recursively follow redirects, only a 200 will resolve.
-          download(response.headers.location, dest).then(() => resolve());
-        } else {
-          reject(
-            `Server responded with ${response.statusCode}: ${response.statusMessage}`,
-          );
-        }
-      });
-      request.on('error', (err) => {
-        reject(err.message);
-      });
+      if (url.startsWith('https')) {
```

Comment:
I'm confused. Why would you execute your request any differently whether it's http or https (apart from any additional verification needed for https)? It makes me think the test for which block to execute is wrong. At minimum, maybe this needs to be extracted to a well-named variable to indicate _why_ you're doing two pretty different things.

More likely, I'm guessing you should extract a common set of instructions to a function and execute the same either way, just using `http` vs `https`

Reply:
done